### ISSUE TITLE
LSTM poor prediction : big gap between loss and val loss

### ISSUE THREAD

**Issue by reyhanhgh** [rstudio/keras#1031]
I am training an lstm model that I constructed using Keras in R following the time series [forecasting tutorial ](https://blogs.rstudio.com/tensorflow/posts/2017-12-20-time-series-forecasting-with-recurrent-neural-networks/) by F. Chollet and J. J. Allaire. The repository [LSTM-PREDICTION](https://github.com/reyhanhgh/LSTM-PREDICTION) includes my script along with the dataset I am working with.

The problem that I have is that there is a huge gap between the training loss and the val_loss and they never converge and after some epochs they stop decreasing. The predictions that I get are very poor.

So I tried to adjust the model. I have tried different kinds of adaptations that had been proposed for similar problems, but with any success. Here is the list of the things that I have tried :
-test different train/validation/test split rates
-test different loss functions
-add/remove activation functions
-test different lookback and sampling step values
-test different optimizers
-test different momentum and decay values (while using sgd), clipping
-test different lr (learning rate) from 1e-6 to 0.1 : I finally saw that a lr value close to the default lr worked better.
-test different step per epochs value
-increase and decrease batch size
-test different structures with different degrees of complexity: in fact adding neither nodes nor layers improved predictions, so I kept the simple initial structure that I had. I also tried to use regularizers : recurrent drop out, drop out, initializers, ..

I know that I have to reduce the training loss (mae) at very early epochs to a level of 0.05 to get decent results. I would highly appreciate if anyone could suggest me a way to achieve this. Many thanks in advance for your time !

**Comment by henry090** [comment]
Hi @reyhanhgh , sorry for late reply. Did you see this tutorial on ```tensorflow.org``` ? It might help you. https://www.tensorflow.org/tutorials/structured_data/time_series

**Comment by reyhanhgh** [comment]
Hi @henry090. No I had not seen this tutorial. Thanks. I’ll check it and let you know if the problem is fixed.
